statistics/handle/handle_hist.go

```diff
@@ -330,6 +345,22 @@ func (h *Handle) writeToChanWithTimeout(taskCh chan *NeededColumnTask, task *Nee
 	return nil
 }
 
+// writeToResultChan safe-writes with panic-recover so one write-fail will not have big impact.
+func (h *Handle) writeToResultChan(resultCh chan model.TableColumnID, rs model.TableColumnID) {
+	defer func() {
+		if r := recover(); r != nil {
+			buf := make([]byte, 4096)
+			stackSize := runtime.Stack(buf, false)
+			buf = buf[:stackSize]
+			logutil.BgLogger().Error("writeToResultChan panicked", zap.Any("error", r), zap.String("stack", string(buf)))
+		}
+	}()
```

Comment:
What kinds of panic will be caught here?

Reply:
Since channel closing is removed, I think no. But still add recover here in case any accident which is out of our understanding about go channel... Because we are writing to multiple resultCh of multiple SQLs at the same time, panic of one should not impact others.